File: src/main/java/org/jukeboxmc/form/element/Toggle.java
package org.jukeboxmc.form.element;


import com.nimbusds.jose.shaded.json.JSONObject;

/**
 * @author GoMint
 * @version 1.0
 */
public class Toggle extends Element {

    private boolean value;

    public Toggle( String id, String text, boolean value ) {
        super( id, text );
        this.value = value;
    }

    @Override
    public JSONObject toJSON() {
        JSONObject obj = super.toJSON();
        obj.put( "type", "toggle" );
        obj.put( "default", this.value );
        return obj;
    }

    @Override
    public Object getAnswer( Object answerOption ) {
        boolean answer = (boolean) answerOption;
        this.value = answer;
        return answer;
    }

}

File: JukeboxMC-API/src/main/kotlin/org/jukeboxmc/api/form/element/Toggle.kt
package org.jukeboxmc.api.form.element

import com.google.gson.JsonObject

/**
 * @author Kaooot
 * @version 1.0
 */
class Toggle(id: String, text: String, private var value: Boolean) : Element(id, text) {

    override fun toJSON(): JsonObject {
        val jsonObject = super.toJSON()
        jsonObject.addProperty("type", "toggle")
        jsonObject.addProperty("default", this.value)
        return jsonObject
    }

    override fun getAnswer(answerOption: Any): Any {
        val answer = super.getAnswer(answerOption) as Boolean
        this.value = answer
        return answer
    }
}